### PR TITLE
Switch to etcd and fix vendoring

### PR DESCRIPTION
**Pull request by kshlm**
glide itself has issues, so now we pull in a huge amount of dependencies. I've
opened issues on glide to help/solve this problems.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,7 +26,7 @@ install:
 
 vendor-update:
 	@echo Updating vendored packages
-	@GO15VENDOREXPERIMENT=1 glide -q up
+	@GO15VENDOREXPERIMENT=1 glide install
```
Comment (atinmu) [inline Makefile]: If the package is already installed, does option 'install' updates it? What's the reason of removing 'up' ?

**Comment by atinmu** [comment]
So do we have the data in terms of dependencies for using libkv/etcd vs etcd-client?

**Comment by kshlm** [comment]
> So do we have the data in terms of dependencies for using libkv/etcd vs etcd-client?

Glide has its own problems so, we don't have a comparison yet.

File: Makefile (hunk 1 above)
Comment (kshlm) [inline Makefile]: With glide-0.8.0, a new `glide.lock` file is created. `glide install` will install/update/downgrade packages listed in the lockfile.

**Pull request merged**